packages/venia-ui/lib/components/CartPage/ProductListing/product.css

```diff
@@ -2,10 +2,20 @@
     align-items: start;
     display: grid;
     grid-gap: 0.5rem 1rem;
-    grid-template-areas: 'image details kebab';
+    grid-template-areas:
+        'error error error'
```

Comment:
It would be nice if empty rows collapsed the gap too, but this unfortunately leads to some extra padding when no error is rendered.

This is me removing the `error error error` row from the grid template:
![](https://i.gyazo.com/b58daa8df630af587171e94cffe4a78c.gif)

To accommodate this I believe we instead always render the error container, but add an `:empty` style to display none, which will correctly collapse that gap.

Reply:
Ok - we should do the same for out of stock then error then :D

Reply:
One option would be to have an error div outside of the root component so we wouldn't need to even worry about grid.

Reply:
I was able to handle this by adding a wrapping element around the entire component with children being the error span and what was previously the product item `li`. The `display:none` is working as expected!

![](https://i.gyazo.com/5bae883831d56d9fc81054c39a9b6f99.png)9a9b6f99.png